index.md

```diff
@@ -8,6 +8,427 @@ This is some welcome text.
+
+    @type lst: list
+    @rtype: int
+    """
+    temp = 0
+    for item in lst:
+        temp += item
+    return temp
+```
+
+### W0104 Pointless statement {#W0104}
+
+This error occurs when a statement doesn't have any effect. This may not be 
+what was intended, so please check your code carefully. 
+
+```python
+def add(lst):
+    """
+    @type lst: list
+    @rtype: int
+    """
+    temp = 0
+    for item in lst:
+        temp += item
+    temp # Error on this line
+```
+
+### W0105 Pointless string statement {#W0105}
+
+This error occurs when string statement has no effect. This may not be 
+what was intended, so please check your code carefully. This error is a special 
+case of the error W0104.
+
+```python
+def letters(lst):
+    """
+    @type lst: list
+    @rtype: int
+    """
+    temp = ""
+    for letter in lst:
+        temp += letter + " "
+    "end" # Error on this line
+```
+
+### W0106 Expression not assigned {#W0107}
+
+This error occurs when an expression that is not a function call is not 
+assigned to anything. If you are getting this error, please check your code 
+carefully.
+ 
+ ```python
+ def add(lst):
+    """
+    @type lst: list
+    @rtype: int
+    """
+    temp = 0
+    for item in lst:
+        temp += item
+    5 + 2 # Error on this line
+```
+
+### W0107 Unnecessary pass {#W0107}
+
+This error occurs when a "pass" statement is used that can be avoided (or has 
+no effect).
+
+```python
+def add(lst):
+    """
+    @type lst: list
+    @rtype: int
+    """
+    temp = 0
+    for item in lst:
+        temp += item
+        pass # Error on this line
+    return temp
+```
+
+### W0109 Duplicate key {#W0109}
+
+This error occurs when a dictionary expression binds the same key multiple 
+times.
+
+```python
+def check(obj):
+    """
+    @type obj: dict
+    @rtype: bool
+    """
+    ex = {'runner1': '5km', 'runner1': '7km'}
+    return ex == obj
+```
+
+### W0125 Using constant test {#W0125}
+
+This error occurs when a conditional statement like an "if" statement uses a 
+constant value for its test. If this error occurs, please check your code 
+carefully.
+
+```python
+def pos():
+    """
+    @rtype: bool
+    """
+    if 0: # Error on this line
+        return False
+    else:
+        return True
+```
+
+### W0141 Bad builtin {#W0141}
+
+This error occurs when a black listed built-in function such as map, filter, 
+input and so on are used. Python now offers some cleaner alternatives like 
+list comprehension. Students are expected to use only the features covered in 
+lectures, or the ones we have explicitly mentioned for an 
+exercise/lab/assignment.
+
+```python
+def pos(obj):
+    """
+    @type obj: int
+    @rtype: bool
+    """
+    if obj < 0:
+        return False
+    else:
+        return True
+
+if __name__ == '__main__':
+    filter(pos, [1, -4, 5]
+```
+
+### W0199 Assert on tuple {#W0199}
+
+This error occurs when an "assert" statement is called like assert (x, y). 
```

Comment:
The pylint documentation here is information - `assert` acting on a tuple always returns true if the tuple is non-empty, and false if it is empty.

Useful to suggest an alternative more explicitly, saying something like "If you want to assert that two expressions are true, then. you should use..."